### ISSUE TITLE
Rotating log generation

### ISSUE THREAD

**Issue by Mrzhuzy** [gabime/spdlog#1321]
After the set time, no new log files are automatically generated.

my  code  ：
      std::vector<spdlog::sink_ptr> sinks;

      sinks.push_back(std::make_shared<spdlog::sinks::stdout_sink_mt>());

      sinks.push_back(std::make_shared<spdlog::sinks::daily_file_sink_mt>(m_plogconfig.filename, 23, 59));
      m_clLogger = std::make_shared<spdlog::logger>(m_plogconfig.logname, begin(sinks), end(sinks));
    
      m_clLogger->set_level((spdlog::level::level_enum)m_plogconfig.level);
      spdlog::flush_every(std::chrono::seconds(m_plogconfig.flushtime));
 
      m_clLogger->flush_on(spdlog::level::level_enum::err);
	    spdlog::register_logger(m_clLogger);

**Comment by tt4g** [comment]
`spdlog::sinks::daily_file_sink` rotates the file before writing log to the current log file buffer.

`spdlog::sinks::daily_file_sink::sink_it_()` writes data to the buffer and performs file rotation if satisfy a condition.
`spdlog::sinks::daily_file_sink::sink_it_()` is called when writing a log with `spdlog::logger`.

`spdlog::logger::flush()` writes buffer data to the log file and consumes the buffer data.
In other words, `spdlog::sinks::daily_file_sink::sink_it_()` is not called.

Your snippet is not rotating the file because it does not write a log.
